### PR TITLE
Rewrite data export.

### PR DESCRIPTION
**Pull request by cmkohnen**
Rewrite data export to work with GFile. Also move logic outside of file_io, since we already distinguish between single item and multiple in the caller function. This no longer uses `numpy.savetxt` since we always expect two-dimensional tuples for values thus reducing a lot of overhead intruduced by mentioned function.

Also refactor some logic.

### CODE REVIEW

File: src/file_io.py [hunk 1]
```diff
@@ -44,23 +43,17 @@ def read_project(file):
...
-        for item in self.datadict.values():
-            xdata = item.xdata
-            ydata = item.ydata
-            filename = item.name.replace("/", "")
-            array = numpy.stack([xdata, ydata], axis=1)
-            file_path = f"{path}/{filename}.txt"
-            if os.path.exists(file_path):
-                file_path = f"{path}/{filename} (copy).txt"
-            numpy.savetxt(str(file_path), array, delimiter="\t")
+def save_file(file, xdata, ydata):
```
Comment (Sjoerd1993) [inline src/file_io.py]: Much neater, nice improvment.

File: src/ui.py [hunk 2]
```diff
@@ -50,34 +52,33 @@ def on_confirm_discard_response(_dialog, response, self):
...
+            for item in self.datadict.values():
+                file = directory.get_child_for_display_name(f"{item.name}.txt")
+                file_io.save_file(file, item.xdata, item.ydata)
         else:
-            path = dialog.save_finish(response).peek_path()
-        file_io.save_file(self, path)
-    except GLib.GError:
-        pass
+            file = dialog.save_finish(response)
+            # return value is not subscriptable, so we loop over one item
```
Comment (Sjoerd1993) [inline src/ui.py]: Not sure which is better, but in principle you can convert the values object to a list with list(self.datadict.values), which is iterable.

The loop can then be replaced by a single item = list(self.datadict.values())[0]

Comment (cmkohnen) [reply src/ui.py]: Ah, i trie `item = set(self.datadict.values())[0]` and still got the non-subscriptable error. This is probably a better solution, as the datadict *might* change between invoking the dialog and accepting it.

**Comment by Sjoerd1993** [comment]
Biggest change in the latest commit seems to be to create and return the item on importing, instead of retreiving the xdata and ydata on import, and then adding that to an item in the gateway function. I'm inclined to agree with that design  :+1:

**Comment by cmkohnen** [comment]
> Biggest change in the latest commit seems to be to create and return the item on importing, instead of retreiving the xdata and ydata on import, and then adding that to an item in the gateway function. I'm inclined to agree with that design +1

Yes, see my commit message. The idea was that maybe sometime a filetype would need to set more properties than just xdata and ydata (e. g. color), so having the item accessible there would remove a lot of refactoring down the line

**Review by Sjoerd1993 — APPROVED** [review]

**Pull request merged**